### DISCUSSION TITLE
Gauging interest in modifier-enbled multi-key events for A11y.

### DISCUSSION THREAD

**Discussion by Alaricus** [clauderic/dnd-kit#910] [Ideas]
Good afternoon, everyone!

The default behavior (marking an item for movement with Space of Enter) works fine so long as those keys aren't reassigned to something else. In case each item is a tile or a link, where Space and Enter are used to interact with it, another key can be specified, but it seems like a combination of keys cannot. I'm working on a codebase where we'd like to mark items for movement with Alt+Shift. Currently it doesn't seem possible to do so, because `activators` check against the list of keys like this: `keyboardCodes.start.includes(code)` and the `keyboardCodes.start` is an array of strings (KeyboardSensor.ts : 287).

I was thinking of making some changes so that `keyboardCodes.start` would contain an array of either strings or objects. If an array element would be an object, it would contain the key (if any) as well as desired statuses for modifies keys. So if the user was to press Q+Alt+Shift, it would look something like this:

```
{
  key: KeyQ,
  altKey: true,
  ctrlKey: false,
  shiftKey: true
}
```
In turn, the variables that are destructured from the `nativeEvent` (KeyboardSensor.ts : 285) would also include `altKey`, `shiftKey`, and `ctrlKey`, which would be then used to compare against the passed object.

Alternatively, arrays could be used instead of objects, so the above could be represented like this:

```
  ["KeyQ", true, false, true]
```
or, in case where only the modifier keys need to be used, like this:
```
  [null, true, false, true]
```

Is this a type of contribution that the maintainers (and the users) would be interested in seeing?

Thank you!

**Comment by Alaricus** [comment]
@clauderic, is this something you would consider accepting?

**Comment by Keenan-Allgood** [comment]
Combing through the discussion and happy to find this discussion has already been started. I have also been looking for some functionality with the alt, shift, and control keys. More specifically the Alt+Shift combination. I have similar code to the proposed setup above in a local fork. @Alaricus, with your proposed changeset, does this also apply to the End (KeyboardSensor.ts: 91) and Cancel(KeyboardSensor.ts: 96) functions?

**Reply by Alaricus** [reply]
I was thinking only in terms of the modifier keys that are already available on the standard KeyboardEvent (the keydown event in this case), which include Alt, Ctrl, and Shift. This ought to make adding this feature fairly easy.

**Reply by Keenan-Allgood** [reply]
Right. I was wondering more if this was planned to be the same for the Start, End, and Cancel Keyboard Codes. For example, you would press Alt+Shift to fire Start/Lift Element, arrow keys to move, and then Alt+Shift to Drop the Element.

Do you have a changeset that is PR ready?

**Reply by Alaricus** [reply]
Sorry, I don't have anything yet, since we solved it another way, and I didn't want to spend time on this until I heard from the maintainer on whether he'd be interested or not.
